### PR TITLE
Update README to explain versioning better

### PR DESCRIPTION
**Pull request by lavalamp**
Trying to clear up more confusion.

### CODE REVIEW

**Comment by lavalamp** [comment]
I updated the table with what I just learned about the 2.0 branch.

File: README.md [hunk 1]
```diff
@@ -26,34 +26,87 @@ Go clients for talking to a [kubernetes](http://kubernetes.io/) cluster.
...
 
-The master branch will track the head in the main Kubernetes repo and accumulating changes. We will make a new client-go major/minor/patch version when:
-* A new major (very rare) or a new minor version is released in the Kubernetes main repository.
-* A feature or a bug fix in the master branch is popular and users want it in a stable branch or with a stable tag. 
+A consequence of this is that `client-go` version numbers will be unrelated to
+Kubernetes version numbers.
+
+#### Branches and tags.
+
+We will create a new branch and tag for each increment in the major version number or
```
Comment (caesarxuchao) [inline README.md]: if we have both release-2.0 and release-2.1 branches, are we going to keep maintaining release-2.0 branch, or will we say bug fixes will only be cherry-picked to release-2.1?

File: README.md [hunk 2]
```diff
@@ -26,34 +26,87 @@ Go clients for talking to a [kubernetes](http://kubernetes.io/) cluster.
...
-We do not back-port new Kubernetes features into older clients. If you need a new feature, you are expected to upgrade to the new client. You can check out the [CHANGELOG](./CHANGELOG.md) for notable changes.
 
 #### Compatibility matrix
 
-* **client-go/1.4** is compatible with Kubernetes 1.3 through 1.5; it includes all features provided by Kubernetes 1.4.
-* **client-go/1.5** is compatible with Kubernetes 1.3 through 1.5; it includes all features provided by Kubernetes 1.4.
+|                | Kubernetes 1.3 | Kubernetes 1.4 | Kubernetes 1.5 | Kubernetes 1.6 (not released yet) |
+|----------------|----------------|----------------|----------------|----------------|
+| client-go 1.4  | +              | ✓              | -              | -              |
+| client-go 1.5  | +              | +              | ✓              | -              |
```
Comment (caesarxuchao) [inline README.md]: client-go 1.5 doesn't match all kubernetes 1.5 features, for example, client-go 1.5 has apps/v1alpha1 (https://github.com/kubernetes/client-go/tree/release-1.5/1.5/pkg/apis/apps/v1alpha1), but kubernetes release-1.5 has [apps/v1beta1](https://github.com/kubernetes/kubernetes/tree/release-1.5/pkg/apis/apps/v1beta1). I'll send a PR.

File: README.md (hunk 1 above)
Comment (lavalamp) [inline README.md]: OK, we can revisit this in the future, but this explains the current 1.4, 1.5 and 2.0 branches.

**Pull request merged**